From 8abcbe9b4cf7eb49f6e1aacf2ee3a45dae4360ba Mon Sep 17 00:00:00 2001
From: classabbyamp <void@placeviolette.net>
Date: Sat, 4 May 2024 15:50:38 -0400
Subject: [PATCH] core-services/03-filesystems.sh: ignore fsck result of 8

exit code of 8 indicates "operational error", which can happen if
CONFIG_BLK_DEV_WRITE_MOUNTED is disabled on linux>=6.8, so ignore it

see also: void-linux/void-packages#50124
---
 core-services/03-filesystems.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core-services/03-filesystems.sh b/core-services/03-filesystems.sh
index bb7eac0..546927c 100644
--- a/core-services/03-filesystems.sh
+++ b/core-services/03-filesystems.sh
@@ -68,7 +68,8 @@ done
 if [ -z "$FASTBOOT" ]; then
     msg "Checking filesystems:"
     fsck -A -T -a -t noopts=_netdev $FORCEFSCK
-    if [ $? -gt 1 ]; then
+    ret="$?"
+    if [ "$ret" -ne 0 ] && [ "$ret" -ne 8 ]; then
         emergency_shell
     fi
 fi
